secret_store/src/key_server_cluster/io/message.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::io::Cursor;
+use std::u16;
+use std::ops::Deref;
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use serde_json;
+use ethcrypto::ecdh::agree;
+use ethkey::{Public, Secret};
+use key_server_cluster::Error;
+use key_server_cluster::message::{Message, ClusterMessage, EncryptionMessage, DecryptionMessage};
+
+/// Size of serialized header.
+pub const MESSAGE_HEADER_SIZE: usize = 4;
+
+#[derive(Debug, PartialEq)]
+/// Message header.
+pub struct MessageHeader {
+	/// Message/Header version.
+	pub version: u8,
+	/// Message kind.
+	pub kind: u8,
+	/// Message payload size (without header).
+	pub size: u16,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+/// Serialized message.
+pub struct SerializedMessage(Vec<u8>);
+
+impl Deref for SerializedMessage {
+	type Target = Vec<u8>;
```

Comment:
ought to be `type Target = [u8]` as most generic code expects this and you don't gain anything but `.capacity()` from `Vec<u8>`